File: app/src/main/java/io/mgba/Data/Database/DAOs/GameDAO.java
package io.mgba.Data.Database.DAOs;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import io.mgba.Data.Database.Game;

@Dao
public interface GameDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Game... games);

    @Update
    void update(Game... games);

    @Delete
    void delete(Game... games);

    @Query("DELETE FROM Games")
    void deleteAll();

    @Query("SELECT * FROM Games")
    List<Game> getGames();

    @Query("SELECT * FROM Games WHERE name LIKE :query")
    List<Game> getGames(String query);

    @Query("SELECT * FROM Games WHERE favourite = 1")
    List<Game> getFavouritesGames();

    @Query("SELECT * FROM Games WHERE platform = :platform")
    List<Game> getGamesForPlatform(int platform);

}

File: app/src/main/java/io/mgba/data/local/database/daos/GameDAO.kt
package io.mgba.data.local.database.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Transaction
import androidx.room.Update
import io.mgba.data.local.database.model.Game

@Dao
interface GameDAO {

    @get:Query("SELECT * FROM Games")
    val games: List<Game>

    @get:Query("SELECT * FROM Games WHERE favourite = 1")
    val favouritesGames: List<Game>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    @Transaction
    fun insert(vararg games: Game)

    @Update
    @Transaction
    fun update(vararg games: Game)

    @Delete
    @Transaction
    fun delete(vararg games: Game)

    @Query("DELETE FROM Games")
    @Transaction
    fun deleteAll()

    @Query("SELECT * FROM Games WHERE name LIKE :query")
    fun getGames(query: String): List<Game>

    @Query("SELECT * FROM Games WHERE platform = :platform")
    fun getGamesForPlatform(platform: Int): List<Game>

}
